### ISSUE TITLE
[bug] Reference-style Links don't work

### ISSUE THREAD

**Issue by TechQuery** [jackey8616/hexo-generator-slidehtml#3]
**Reference-style Links**

```markdown
[text][1]

 [1]: http://example.com
```

will be displayed as _pure text_ `[text][1]`...

**Comment by jackey8616** [comment]
@TechQuery  
Hi there.
Can you provide more detail content of your article which want to generate slide?
I added an article to test the reference link you mention above and it's fine.
But only reference link write in same page would work.
Here is the [link](http://www.clo5de.info/hexo-generator-slidehtml/2019/04/03/referenceTitleTest/) [slide-link](http://www.clo5de.info/hexo-generator-slidehtml/2019/04/03/referenceTitleTest/slide.html).

source is here
```html
---
title: Reference  Title Test
slidehtml:
  titleMerge: false
---

# This is Title

--

[text][1]

[1]: http://example.com

--
```

**Comment by TechQuery** [comment]
Yes, it'll be failed if all the Reference Links are located at bottom of the whole file.

**Comment by jackey8616** [comment]
@TechQuery 
Basically, I would suggest you to put reference links alone in each slide bottom when need it.
If you want to reach some code like below, it would be very difficult and unnecessary.
```
[link][1]

--

[link2][2]

--

[1]: http://example.com
[2]: http://example2.com
```

Here is two reasons that I do not want to put in bottom slide:
- 1. It's hard to trace every links when you got a massive slides.
- 2. Putting links at bottom slide would cause slide-logic more complex.
      It needs more logic to decides the slide should display or not, since that reference links should not appear. More further, we need to directly access the raw-markdown code in the worst situation.

Let me know, if you have any cool idea.
Also welcome to PR if you really want this feature.

**Comment by TechQuery** [comment]
You split MarkDown code fragments first, then transform them to slides ?

We may convert the whole MarkDown file to HTML, after that, render the sections to slides.

**Comment by jackey8616** [comment]
Raw-markdown will be put in a ejs template and parse by reveal.js plugin.
This repo is only the bridge of hexo-plugin and reveal.js.
It doesn't responsible for any parse.

**Comment by TechQuery** [comment]
`{% asset_img example.png %}` will be failed, too...

I think we should parse **Raw Markdown** with Hexo utility methods before Slides rendering.
